pkg/blockbuilder/blockbuilder.go

```diff
@@ -867,54 +848,49 @@ func (b *BlockBuilder) consumePartition(
 			Offset: firstRec.Offset,
 		}
 		b.seekPartition(client, part, rec)
-		return pl, nil
+		return nil
 	}
 
 	// All samples in all records were processed. We can commit the last record's offset.
 	if commitRec == nil {
 		commitRec = lastRec
```

Comment:
Just noticed that if all samples faced tsdb soft error, `lastRec` will be nil. We should avoid panics and probably skip commits if lastRec and commitRec both are nil.


But this may cause BB to get stuck in replaying the same data, in which case committing the last record may make sense even if TSDB rejected it. But avoiding panics is better than trying again.

Reply:
Isn't that what we handle below https://github.com/grafana/mimir/blob/9280ea7fadb2a30201b395f33aae59b68021a9e3/pkg/blockbuilder/blockbuilder.go#L809-L818 ?